lib/ansible/modules/cloud/amazon/aws_sgw_facts.py

```diff
@@ -0,0 +1,358 @@
+        for gw in aws_response["Gateways"]:
+            gateway_list.append(camel_dict_to_snake_dict(gw))
+        return aws_response["Marker"] if "Marker" in aws_response else None
+
+    """
+    List all storage gateways for the AWS endpoint.
+    """
+    def list_gateways(self):
+        response = self.connection.list_gateways(
+            Limit=100
+        )
+
+        gateways = []
+        marker = self._read_gatewaylist_reponse(gateways, response)
+
+        while marker is not None:
+            response = self.connection.list_gateways(
+                Limit=100,
+                Marker=marker
+            )
+            marker = self._read_gatewaylist_reponse(gateways, response)
+
+        return gateways
+
+    """
+    Read fileshare objects from AWS API response.
+    Drop the gateway_arn attribute from response, as it will be duplicate with parent object.
+    """
+    @staticmethod
+    def _read_gateway_fileshare_response(fileshares, aws_reponse):
+        for share in aws_reponse["FileShareInfoList"]:
+            share_obj = camel_dict_to_snake_dict(share)
+            if "gateway_arn" in share_obj:
+                del share_obj["gateway_arn"]
+            fileshares.append(share_obj)
+
+        return aws_reponse["NextMarker"] if "NextMarker" in aws_reponse else None
+
+    """
+    List file shares attached to AWS storage gateway when in S3 mode.
+    """
+    def list_gateway_file_shares(self, gateway):
+        response = self.connection.list_file_shares(
+            GatewayARN=gateway["gateway_arn"],
+            Limit=100
+        )
+
+        gateway["file_shares"] = []
+        marker = self._read_gateway_fileshare_response(gateway["file_shares"], response)
+
+        while marker is not None:
+            response = self.connection.list_file_shares(
+                GatewayARN=gateway["gateway_arn"],
+                Marker=marker,
+                Limit=100
+            )
+
+            marker = self._read_gateway_fileshare_response(gateway["file_shares"], response)
+
+    """
+    List storage gateway local disks
+    """
+    def list_local_disks(self, gateway):
+        response = self.connection.list_local_disks(
+            GatewayARN=gateway["gateway_arn"]
+        )
+
+        gateway["local_disks"] = []
+        for disk in response["Disks"]:
+            gateway["local_disks"].append(camel_dict_to_snake_dict(disk))
+
+    """
+    Read tape objects from AWS API response.
+    Drop the gateway_arn attribute from response, as it will be duplicate with parent object.
+    """
+    @staticmethod
+    def _read_gateway_tape_response(tapes, aws_response):
+        for tape in aws_response["TapeInfos"]:
+            tape_obj = camel_dict_to_snake_dict(tape)
+            if "gateway_arn" in tape_obj:
+                del tape_obj["gateway_arn"]
+            tapes.append(tape_obj)
+
+        return aws_response["Marker"] if "Marker" in aws_response else None
```

Comment:
Use a paginator: boto3.readthedocs.io/en/latest/reference/services/storagegateway.html#paginators

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#api-throttling-and-pagination

Reply:
Might be worth raising bugs against botocore where paginators don't exist but should. Won't help us now but we can migrate to them a year after the bugs are resolved ;)